### PR TITLE
Dockerfile

### PR DESCRIPTION
**Pull request by pablodeymo**
# Dockerfile

## Description

The goal of this PR is to have a Dockerfile with all the dependencies needed.

## Checklist
- [ ] Linked to Github Issue
- [ ] Unit tests added
- [ ] Integration tests added.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+# Install rust nightly-2023-06-19 (1.72.0-nightly)
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | bash -s -- -y --default-toolchain=nightly-2023-06-19
+ENV PATH="/root/.cargo/bin:${PATH}"
+
+# Copy cairo_native code
+COPY . /cairo_native/
+
+# Fetch cairo corelibs
+RUN git clone --depth 1 \
+    --branch v1.1.0 \
```
Comment (edg-l) [inline Dockerfile:30]: the branch needs to be updated
```suggestion
    --branch v2.0.1 \
```

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    https://github.com/starkware-libs/cairo.git \
+    starkware-cairo
+RUN cp -r starkware-cairo/corelib /cairo_native
+RUN rm -rf starkware-cairo/
+
+# Compile cairo_native
+WORKDIR /cairo_native/
+ENV MLIR_SYS_160_PREFIX=/usr/lib/llvm-16
+RUN cargo +nightly-2023-06-19 build --release
```
Comment (edg-l) [inline Dockerfile:39]: Use this (from make build) to build with all the features enabled.
```suggestion
RUN cargo +nightly-2023-06-19 build --release --all-features
```

**Review by edg-l — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (azteca1998) [reply Dockerfile:30]: Or invoke the script which does that (the one ni `./scripts/fetch-corelibs.sh`)

File: Dockerfile (hunk 2 above)
Comment (azteca1998) [reply Dockerfile:39]: Or use the `Makefile` like this: `make build`.

**Review by azteca1998 — APPROVED** [review]

**Review by edg-l — APPROVED** [review]

**Pull request merged**